src/backend/distributed/utils/resource_lock.c

```diff
@@ -192,6 +193,31 @@ LockShardListResources(List *shardIntervalList, LOCKMODE lockMode)
 
 
 /*
+ * LockRelationShards takes locks on all shards in a list of RelationShards
+ * to prevent concurrent DML statements on those shards.
+ */
+void
+LockRelationShardListResources(List *relationShardList, LOCKMODE lockMode)
```

Comment:
Was asked to change `LockShardResources` to `LockShardListResources` in another PR so it seems slightly inconsistent, but I'm fine with either form.